br/pkg/lightning/backend/external/byte_reader.go

```diff
@@ -98,8 +98,8 @@ func (r *byteReader) switchToConcurrentReaderImpl() error {
 	r.conReader.currentFileOffset = currOffset
 	r.conReader.bufferReadOffset = 0
 
-	r.useConcurrentReaderCurrent.Store(true)
 	r.conReader.buffer = make([]byte, r.conReader.concurrency*r.conReader.readBufferSize)
+	r.useConcurrentReaderCurrent.Store(true)
```

Comment:
In fact, the use of switchToConcurrentReaderImpl is sequential. So there is no other goroutine that would access the buffer. However, the atomic variable can be refined to a normal boolean.